### PR TITLE
Ignore files that shouldn't be tracked

### PR DESCRIPTION
**Pull request by tcmal**
Add files generated by `npm run tester` to `.gitignore` as they are redundant.

## Description

  - Ignore files in `test/` that end in `.js`, except those that end in `.test.js`
  - Remove files that are now ignored by this rule that previously weren't
  - Remove some old test folders/cases that no longer have snippets attached to them
     - Presumably incomplete removals

## What does your PR belong to?
- [ ] Website
- [ ] Snippets
- [X] General / Things regarding the repository (like CI Integration)
- [X] Tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (non-breaking improvement of a snippet)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have checked that the changes are working properly
- [X] I have checked that there isn't any PR doing the same
- [X] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by Chalarangelo** [comment]
@kingdavidmartins Do these changes have the potential to break our build and test process? If not, I'm fine with merging the PR.

**Comment by kingdavidmartins** [comment]
@Chalarangelo I'm personally not up for ignoring `test/**/*.js` as It will do the following

- breaks `npm test` for potential contributors who fork the project & haven't ran `npm run tester`. So anyone who wants work/play around with the `test/` dir has to run `npm run tester` 1st.

- Makes adding test a tad bit harder for a beginner, and a tad bit annoying for contributors/developers who might want to add test on the browser through github web app. Since if a snippet gets updated and a beginner/contributor/developer want to know what the snippets does. They won't be able to just look at the `snippetName.js` file in the current dir. They would now have to navigate to the `README.md` or the `snippets/snippetName.md`

<strong>*Note:*</strong> *Although not serious repercussions. Why give up whatever little benefits we can offer to our users for nothing in return?*

I believe ignoring `test/**/*.js` doesn't directly add any value to the project. Ignoring `test/**/*.js` actually has the potential to not only remove our current benefits. But the possibility of adding complexity  and/or extra processes to the project. Adding tests

File: test/README/README.test.js [hunk 1]
```diff
@@ -1,13 +0,0 @@
...
-test('Testing README', (t) => {
-  //For more information on all the methods supported by tape
-  //Please go to https://github.com/substack/tape
-  t.true(typeof README === 'function', 'README is a Function');
-  //t.deepEqual(README(args..), 'Expected');
-  //t.equal(README(args..), 'Expected');
-  //t.false(README(args..), 'Expected');
-  //t.throws(README(args..), 'Expected');
-  t.end();
-});
```
Comment (kingdavidmartins) [inline test/README/README.test.js]: Nice! 👍

Comment (kingdavidmartins) [inline test/README/README.test.js]: 👍

File: test/isArray/isArray.test.js [hunk 2]
```diff
@@ -1,14 +0,0 @@
...
-  //For more information on all the methods supported by tape
-  //Please go to https://github.com/substack/tape
-  t.true(typeof isArray === 'function', 'isArray is a Function');
-  t.equal(isArray([1]), true, "passed value is an array");
-  t.equal(isArray('array'), false, "passed value is not an array");
-  //t.equal(isArray(args..), 'Expected');
-  //t.false(isArray(args..), 'Expected');
-  //t.throws(isArray(args..), 'Expected');
-  t.end();
-});
```
Comment (kingdavidmartins) [inline test/isArray/isArray.test.js]: 👍

File: test/isMap/isMap.test.js [hunk 3]
```diff
@@ -1,13 +0,0 @@
...
-test('Testing isMap', (t) => {
-  //For more information on all the methods supported by tape
-  //Please go to https://github.com/substack/tape
-  t.true(typeof isMap === 'function', 'isMap is a Function');
-  //t.deepEqual(isMap(args..), 'Expected');
-  //t.equal(isMap(args..), 'Expected');
-  //t.false(isMap(args..), 'Expected');
-  //t.throws(isMap(args..), 'Expected');
-  t.end();
-});
```
Comment (kingdavidmartins) [inline test/isMap/isMap.test.js]: 👍

File: test/isArrayBuffer/isArrayBuffer.test.js [hunk 4]
```diff
@@ -1,13 +0,0 @@
...
-test('Testing isArrayBuffer', (t) => {
-  //For more information on all the methods supported by tape
-  //Please go to https://github.com/substack/tape
-  t.true(typeof isArrayBuffer === 'function', 'isArrayBuffer is a Function');
-  //t.deepEqual(isArrayBuffer(args..), 'Expected');
-  //t.equal(isArrayBuffer(args..), 'Expected');
-  //t.false(isArrayBuffer(args..), 'Expected');
-  //t.throws(isArrayBuffer(args..), 'Expected');
-  t.end();
-});
```
Comment (kingdavidmartins) [inline test/isArrayBuffer/isArrayBuffer.test.js]: 👍

File: test/README/README.test.js (hunk 1 above)
Comment (kingdavidmartins) [inline test/README/README.test.js]: 👍

File: test/isArray/isArray.test.js (hunk 2 above)
Comment (kingdavidmartins) [inline test/isArray/isArray.test.js]: 👍

File: test/isArrayBuffer/isArrayBuffer.test.js (hunk 4 above)
Comment (kingdavidmartins) [inline test/isArrayBuffer/isArrayBuffer.test.js]: 👍

File: test/isWeakSet/isWeakSet.test.js [hunk 5]
```diff
@@ -1,13 +0,0 @@
...
-test('Testing isWeakSet', (t) => {
-  //For more information on all the methods supported by tape
-  //Please go to https://github.com/substack/tape
-  t.true(typeof isWeakSet === 'function', 'isWeakSet is a Function');
-  //t.deepEqual(isWeakSet(args..), 'Expected');
-  //t.equal(isWeakSet(args..), 'Expected');
-  //t.false(isWeakSet(args..), 'Expected');
-  //t.throws(isWeakSet(args..), 'Expected');
-  t.end();
-});
```
Comment (kingdavidmartins) [inline test/isWeakSet/isWeakSet.test.js]: 👍

File: test/isWeakMap/isWeakMap.test.js [hunk 6]
```diff
@@ -1,13 +0,0 @@
...
-test('Testing isWeakMap', (t) => {
-  //For more information on all the methods supported by tape
-  //Please go to https://github.com/substack/tape
-  t.true(typeof isWeakMap === 'function', 'isWeakMap is a Function');
-  //t.deepEqual(isWeakMap(args..), 'Expected');
-  //t.equal(isWeakMap(args..), 'Expected');
-  //t.false(isWeakMap(args..), 'Expected');
-  //t.throws(isWeakMap(args..), 'Expected');
-  t.end();
-});
```
Comment (kingdavidmartins) [inline test/isWeakMap/isWeakMap.test.js]: 👍

File: test/isTypedArray/isTypedArray.test.js [hunk 7]
```diff
@@ -1,13 +0,0 @@
...
-test('Testing isTypedArray', (t) => {
-  //For more information on all the methods supported by tape
-  //Please go to https://github.com/substack/tape
-  t.true(typeof isTypedArray === 'function', 'isTypedArray is a Function');
-  //t.deepEqual(isTypedArray(args..), 'Expected');
-  //t.equal(isTypedArray(args..), 'Expected');
-  //t.false(isTypedArray(args..), 'Expected');
-  //t.throws(isTypedArray(args..), 'Expected');
-  t.end();
-});
```
Comment (kingdavidmartins) [inline test/isTypedArray/isTypedArray.test.js]: 👍

File: test/isSet/isSet.test.js [hunk 8]
```diff
@@ -1,13 +0,0 @@
...
-test('Testing isSet', (t) => {
-  //For more information on all the methods supported by tape
-  //Please go to https://github.com/substack/tape
-  t.true(typeof isSet === 'function', 'isSet is a Function');
-  //t.deepEqual(isSet(args..), 'Expected');
-  //t.equal(isSet(args..), 'Expected');
-  //t.false(isSet(args..), 'Expected');
-  //t.throws(isSet(args..), 'Expected');
-  t.end();
-});
```
Comment (kingdavidmartins) [inline test/isSet/isSet.test.js]: 👍

File: test/isRegExp/isRegExp.test.js [hunk 9]
```diff
@@ -1,13 +0,0 @@
...
-test('Testing isRegExp', (t) => {
-  //For more information on all the methods supported by tape
-  //Please go to https://github.com/substack/tape
-  t.true(typeof isRegExp === 'function', 'isRegExp is a Function');
-  //t.deepEqual(isRegExp(args..), 'Expected');
-  //t.equal(isRegExp(args..), 'Expected');
-  //t.false(isRegExp(args..), 'Expected');
-  //t.throws(isRegExp(args..), 'Expected');
-  t.end();
-});
```
Comment (kingdavidmartins) [inline test/isRegExp/isRegExp.test.js]: 👍

File: test/isMap/isMap.test.js (hunk 3 above)
Comment (kingdavidmartins) [inline test/isMap/isMap.test.js]: 👍

File: test/isArrayBuffer/isArrayBuffer.test.js (hunk 4 above)
Comment (kingdavidmartins) [inline test/isArrayBuffer/isArrayBuffer.test.js]: 👍

File: test/README/README.test.js (hunk 1 above)
Comment (kingdavidmartins) [inline test/README/README.test.js]: 👍

File: test/isArray/isArray.test.js (hunk 2 above)
Comment (kingdavidmartins) [inline test/isArray/isArray.test.js]: 👍

**Comment by Chalarangelo** [comment]
@kingdavidmartins So are we good to merge this, then?

**Comment by kingdavidmartins** [comment]
@Chalarangelo I personally think we shouldn't ignore `test/**/*.js` for reasons stated above. 

Once removed from `.gitignore` and changes reflected. I believe this would be good to go.

**Comment by Chalarangelo** [comment]
@kingdavidmartins I completely missed your original comment, sorry. I agree with that, your argument is quite valid on this issue. I think the easiest way to deal with this is to close the PR and just update the `tdd` script to remove old/removed/invalid test folders and files, so that we keep everything clean, therefore I am closing and the important changes will be made automatically as soon as the script is updated.

**Comment by kingdavidmartins** [comment]
@Chalarangelo Agreed. Should submit a PR updating the `tdd` script later today that fixes said issues

**Pull request closed without merging**